OBSDOCS-1723: Port the "Installing Logging" sections to 5.8 and 6.y docs (#89053)
openshift/openshift-docs: modules/log6x-logging-loki-cli-install.adoc

// Module is included in the following assemblies:
//
// * observability/logging/log_storage/installing-log-storage.adoc

:_mod-docs-content-type: PROCEDURE
[id="log6x-logging-loki-cli-install_{context}"]
= Installing {logging-uc} and the {loki-op} using the CLI

To install and configure logging on your {product-title} cluster, an Operator such as {loki-op} for log storage must be installed first. This can be done from the {product-title} CLI.

.Prerequisites

* You have administrator permissions.
* You installed the {oc-first}.
* You have access to a supported object store. For example: AWS S3, Google Cloud Storage, Azure, Swift, Minio, or {rh-storage}.

.Procedure

--
include::snippets/logging-stable-updates-snip.adoc[leveloffset=+1]
--

. Create a `Namespace` object for {loki-op}:
+
.Example `Namespace` object
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: openshift-operators-redhat # <1>
  annotations:

>     openshift.io/node-selector: ""

Author: removed, should this be removed from 5.8 also?

  labels:
    openshift.io/cluster-monitoring: "true" # <2>
----

> <1> You must specify the `openshift-operators-redhat` namespace. To prevent possible conflicts with metrics, you should configure the Prometheus Cluster Monitoring stack to scrape metrics from the `openshift-operators-redhat` namespace and not the `openshift-operators` namespace. The `openshift-operators` namespace might contain community Operators, which are untrusted and could publish a metric with the same name as an {product-title} metric, which would cause conflicts.

Author: done

<2> A string value that specifies the label as shown to ensure that cluster monitoring scrapes the `openshift-operators-redhat` namespace.

. Apply the `Namespace` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a `Subscription` object for {loki-op}:
+
.Example `Subscription` object
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: loki-operator
  namespace: openshift-operators-redhat # <1>
spec:

>   channel: stable # <2>

Reviewer: We don't publish `stable`
```suggestion
  channel: stable-6.<y> # <2>
```

Author: done

  name: loki-operator
  source: redhat-operators # <3>
  sourceNamespace: openshift-marketplace
----
<1> You must specify the `openshift-operators-redhat` namespace.

> <2> Specify `stable`, or `stable-6.<y>` as the channel.

Reviewer: ```suggestion
<2> Specify `stable-6.<y>` as the channel.
```

Author: done

<3> Specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the `CatalogSource` object you created when you configured the Operator Lifecycle Manager (OLM).

. Apply the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a `namespace` object for the {clo}:
+
.Example `namespace` object
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: openshift-logging # <1>

>   annotations:
>     openshift.io/node-selector: ""
>   labels:
>     openshift.io/cluster-logging: "true"

Reviewer: ```suggestion
  labels:
```

Author: done

----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: cluster-logging
  namespace: openshift-logging # <1>
spec:
  targetNamespaces:
  - openshift-logging
----
<1> You must specify the `openshift-logging` namespace.

. Apply the `OperatorGroup` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a `Subscription` object:
+
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: cluster-logging
  namespace: openshift-logging # <1>
spec:
  channel: stable # <2>
  name: cluster-logging
  source: redhat-operators # <3>
  sourceNamespace: openshift-marketplace
----
<1> You must specify the `openshift-logging` namespace.
<2> Specify `stable`, or `stable-6.<y>` as the channel.
<3> Specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the CatalogSource object you created when you configured the Operator Lifecycle Manager (OLM).

. Apply the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

> 

Reviewer: We are missing the part about creating a LokiStack.

Author: added

// Module is included in the following assemblies:
//
// * observability/logging/log_storage/installing-log-storage.adoc

:_mod-docs-content-type: PROCEDURE
[id="log6x-logging-loki-cli-install_{context}"]
= Installing {logging-uc} by using the CLI

> 

Author: I've renamed this section to Installing the Loki Operator by using the CLI

. Apply the `LokiStack CR` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----
+
--
include::snippets/logging-retention-period-snip.adoc[leveloffset=+1]
--

. Verify the installation by running the following command:
+
[source,terminal]
----
$ oc get pods -n openshift-logging
----
+
.Example output
[source,terminal]
----
$ oc get pods -n openshift-logging
NAME                                               READY   STATUS    RESTARTS   AGE
cluster-logging-operator-fb7f7cf69-8jsbq           1/1     Running   0          98m
collector-222js                                    2/2     Running   0          18m
collector-g9ddv                                    2/2     Running   0          18m
collector-hfqq8                                    2/2     Running   0          18m
collector-sphwg                                    2/2     Running   0          18m
collector-vv7zn                                    2/2     Running   0          18m
collector-wk5zz                                    2/2     Running   0          18m

> logging-view-plugin-6f76fbb78f-n2n4n               1/1     Running   0          18m

Author: fixed

lokistack-sample-compactor-0                       1/1     Running   0          42m
lokistack-sample-distributor-7d7688bcb9-dvcj8      1/1     Running   0          42m
lokistack-sample-gateway-5f6c75f879-bl7k9          2/2     Running   0          42m
lokistack-sample-gateway-5f6c75f879-xhq98          2/2     Running   0          42m
lokistack-sample-index-gateway-0                   1/1     Running   0          42m
lokistack-sample-ingester-0                        1/1     Running   0          42m
lokistack-sample-querier-6b7b56bccc-2v9q4          1/1     Running   0          42m
lokistack-sample-query-frontend-84fb57c578-gq2f7   1/1     Running   0          42m

> ----

Author: Thank you so much for writing this!! I've created a separate procedure for this and added to the installation assembly.

  labels:
    openshift.io/cluster-monitoring: "true" # <2>
----
<1> The Red{nbsp}Hat OpenShift Logging Operator is only deployable to the `openshift-logging` namespace.
<2> A string value that specifies the label as shown to ensure that cluster monitoring scrapes the `openshift-operators-redhat` namespace.

. Apply the `namespace` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create an `OperatorGroup` object
+
.Example `OperatorGroup` object
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: loki-operator
  namespace: openshift-operators-redhat # <1>
spec:
  targetNamespaces:
  - openshift-logging
----
<1> You must specify the `openshift-operators-redhat` namespace.

. Apply the `OperatorGroup` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a `LokiStack` CR:
+
.Example `LokiStack` CR
[source,yaml]
----
apiVersion: loki.grafana.com/v1
kind: LokiStack
metadata:
  name: logging-loki # <1>
  namespace: openshift-logging # <2>
spec:
  size: 1x.small # <3>
  storage:
    schemas:
    - version: v13

>       effectiveDate: "<yyyy>-<mm>-<dd>"
>     secret:
>       name: logging-loki-s3 # <4>
>       type: s3 # <5>
>       credentialMode:
>   storageClassName: <storage_class_name> # <6>
>   tenants:
>     mode: openshift-logging # <7>
> ----
> <1> Use the name `logging-loki`.
> <2> You must specify the `openshift-logging` namespace.
> <3> Specify the deployment size. Supported size options for production instances of Loki are `1x.extra-small`, `1x.small`, or `1x.medium`. Additionally, `1x.pico` is supported starting with {logging} 6.1.
> <4> Specify the name of your log store secret.
> <5> Specify the corresponding storage type.
> <6> Specify the name of a storage class for temporary storage. For best performance, specify a storage class that allocates block storage. Available storage classes for your cluster can be listed by using the `oc get storageclasses` command.
> <7> `openshift-logging` is the default tenancy mode where a tenant will be created for each log type: audit, infrastructure, and application logs. This enables access control for individual users and user groups to different log streams.

Reviewer: ```suggestion
      effectiveDate: "<yyyy>-<mm>-<dd>" # <4>
    secret:
      name: logging-loki-s3 # <5>
      type: s3 # <6>
      credentialMode:
  storageClassName: <storage_class_name> # <7>
  tenants:
    mode: openshift-logging # <8>
----
<1> Use the name `logging-loki`.
<2> You must specify the `openshift-logging` namespace.
<3> Specify the deployment size. Supported size options for production instances of Loki are `1x.extra-small`, `1x.small`, or `1x.medium`. Additionally, `1x.pico` is supported starting with {logging} 6.1.
<4> For new installations this date should be set to the equivalent of "yesterday", as this will be the date from when the schema defined above will take effect.
<5> Specify the name of your log store secret.
<6> Specify the corresponding storage type.
<7> Specify the name of a storage class for temporary storage. For best performance, specify a storage class that allocates block storage. Available storage classes for your cluster can be listed by using the `oc get storageclasses` command.
<8> `openshift-logging` is the default tenancy mode where a tenant will be created for each log type: audit, infrastructure, and application logs. This enables access control for individual users and user groups to different log streams.
```

. Apply the `LokiStack CR` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----
+
--
include::snippets/logging-retention-period-snip.adoc[leveloffset=+1]
--

. Create a secret with the credentials to access the object storage. For example, create a secret to access {aws-first} s3.
+
.Example `Secret` object
[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: logging-loki-s3 <1>
  namespace: openshift-logging
stringData: <2>
  access_key_id: AKIAIOSFODNN7EXAMPLE
  access_key_secret: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
  bucketnames: s3-bucket-name
  endpoint: https://s3.eu-central-1.amazonaws.com
  region: eu-central-1
----
<1> Use the name `logging-loki-s3` to match the name used in LokiStack.
<2> For the contents of the secret see the Loki object storage section.

> 

Reviewer: This one seems like it was missed

+
.Example `Namespace` object
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: openshift-operators-redhat # <1>
  labels:
    openshift.io/cluster-monitoring: "true" # <2>
----
<1> You must specify the `openshift-operators-redhat` namespace. To enable monitoring for the operator, you should configure the Cluster Monitoring Operator to scrape metrics from the `openshift-operators-redhat` namespace and not the `openshift-operators` namespace. The `openshift-operators` namespace might contain community Operators, which are untrusted and could publish a metric with the same name as an {product-title} metric, which would cause conflicts.
<2> A string value that specifies the label as shown to ensure that cluster monitoring scrapes the `openshift-operators-redhat` namespace.

. Apply the `Namespace` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a `Subscription` object for {loki-op}:
+
.Example `Subscription` object
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: loki-operator
  namespace: openshift-operators-redhat # <1>
spec:

>   channel: stable-6.<y> # <2>
>   name: loki-operator
>   source: redhat-operators # <3>
>   sourceNamespace: openshift-marketplace
> ----
> <1> You must specify the `openshift-operators-redhat` namespace.
> <2> Specify `stable-6.<y>` as the channel.
> <3> Specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the `CatalogSource` object you created when you configured the Operator Lifecycle Manager (OLM).

Reviewer: ```suggestion
  channel: stable-6.<y> # <2>
  installPlanApproval: Automatic # <3>
  name: loki-operator
  source: redhat-operators # <4>
  sourceNamespace: openshift-marketplace
----
<1> You must specify the `openshift-operators-redhat` namespace.
<2> Specify `stable-6.<y>` as the channel.
<3> If the approval strategy in the subscription is set to `Automatic`, the update process initiates as soon as a new Operator version is available in the selected channel. If the approval strategy is set to `Manual`, you must manually approve pending updates.
<4> Specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the `CatalogSource` object you created when you configured the Operator Lifecycle Manager (OLM).
```

<2> You must specify the `openshift-logging` namespace.
<3> Specify the deployment size. Supported size options for production instances of Loki are `1x.extra-small`, `1x.small`, or `1x.medium`. Additionally, `1x.pico` is supported starting with {logging} 6.1.
<4> For new installations this date should be set to the equivalent of "yesterday", as this will be the date from when the schema will take effect.
<5> Specify the name of your log store secret.
<6> Specify the corresponding storage type.
<7> Specify the name of a storage class for temporary storage. For best performance, specify a storage class that allocates block storage. Available storage classes for your cluster can be listed by using the `oc get storageclasses` command.
<8> `openshift-logging` is the default tenancy mode where a tenant will be created for each log type: audit, infrastructure, and application logs. This enables access control for individual users and user groups to different log streams.

. Apply the `LokiStack CR` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

--
include::snippets/logging-retention-period-snip.adoc[leveloffset=+1]
--

> . Create a secret with the credentials to access the object storage. For example, create a secret to access {aws-first} s3.

Reviewer: Something broke the enumeration. In this step it get reset to 1.

spec:
  channel: stable-6.<y> # <2>
  installPlanApproval: Automatic # <3>
  name: loki-operator
  source: redhat-operators # <4>
  sourceNamespace: openshift-marketplace
----
<1> You must specify `openshift-operators-redhat` as the namespace.
<2> Specify `stable-6.<y>` as the channel.
<3> If the approval strategy in the subscription is set to `Automatic`, the update process initiates as soon as a new operator version is available in the selected channel. If the approval strategy is set to `Manual`, you must manually approve pending updates.
<4> Specify `redhat-operators` as the value. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the `CatalogSource` object that you created when you configured Operator Lifecycle Manager (OLM).

. Apply the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a `namespace` object for deploy the LokiStack:
+
.Example `namespace` object
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: openshift-logging # <1>
  labels:
    openshift.io/cluster-monitoring: "true" # <2>
----

> <1> `openshift-logging` is the namespace dedicated for all {logging} workloads.

Reviewer 2: ```suggestion
<1> The `openshift-logging` namespace is dedicated for all {logging} workloads.
```

[source,yaml]
----
apiVersion: loki.grafana.com/v1
kind: LokiStack
metadata:
  name: logging-loki # <1>
  namespace: openshift-logging # <2>
spec:
  size: 1x.small # <3>
  storage:
    schemas:
    - version: v13
      effectiveDate: "<yyyy>-<mm>-<dd>" # <4>
    secret:
      name: logging-loki-s3 # <5>
      type: s3 # <6>
  storageClassName: <storage_class_name> # <7>
  tenants:
    mode: openshift-logging # <8>
----
<1> Use the name `logging-loki`.
<2> You must specify `openshift-logging` as the namespace.
<3> Specify the deployment size. Supported size options for production instances of Loki are `1x.extra-small`, `1x.small`, or `1x.medium`. Additionally, `1x.pico` is supported starting with {logging} 6.1.
<4> For new installations this date should be set to the equivalent of "yesterday", as this will be the date from when the schema takes effect.
<5> Specify the name of your log store secret.
<6> Specify the corresponding storage type.
<7> Specify the name of a storage class for temporary storage. For best performance, specify a storage class that allocates block storage. You can list the available storage classes for your cluster by using the `oc get storageclasses` command.

> <8> `openshift-logging` is the default tenancy mode where a tenant will be created for each log type: audit, infrastructure, and application logs. This enables access control for individual users and user groups to different log streams.

Reviewer 2: Same nit as mentioned in the other comment.

> . Apply the `LokiStack CR` object by running the following command:

Reviewer 2: ```suggestion
. Apply the `LokiStack` CR object by running the following command:
```

.Procedure

. Create a `Namespace` object for {loki-op}:
+
.Example `Namespace` object
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: openshift-operators-redhat # <1>
  labels:
    openshift.io/cluster-monitoring: "true" # <2>
----
<1> You must specify `openshift-operators-redhat` as the namespace. To enable monitoring for the operator, configure Cluster Monitoring Operator to scrape metrics from the `openshift-operators-redhat` namespace and not the `openshift-operators` namespace. The `openshift-operators` namespace might contain community operators, which are untrusted and could publish a metric with the same name as an {product-title} metric, causing conflicts.
<2> A string value that specifies the label as shown to ensure that cluster monitoring scrapes the `openshift-operators-redhat` namespace.

. Apply the `Namespace` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create an `OperatorGroup` object.
+
.Example `OperatorGroup` object
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: loki-operator
  namespace: openshift-operators-redhat # <1>
  annotations:

>     olm.providedAPIs: AlertingRule.v1.loki.grafana.com,LokiStack.v1.loki.grafana.com,RecordingRule.v1.loki.grafana.com,RulerConfig.v1.loki.grafana.com

Reviewer 3: The annotations can be removed and they'll be added automatically when the operator is installed successfully.

Author: removed

+
.Example `Secret` object
[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: logging-loki-s3 # <1>
  namespace: openshift-logging
stringData: # <2>
  access_key_id: <access_key_id>
  access_key_secret: <access_secret>
  bucketnames: s3-bucket-name
  endpoint: https://s3.eu-central-1.amazonaws.com
  region: eu-central-1
----
<1> Use the name `logging-loki-s3` to match the name used in LokiStack.
<2> For the contents of the secret see the Loki object storage section.
+
--
include::snippets/logging-retention-period-snip.adoc[leveloffset=+1]
--

. Apply the `Secret` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a `LokiStack` CR:
+
.Example `LokiStack` CR
[source,yaml]
----
apiVersion: loki.grafana.com/v1
kind: LokiStack
metadata:
  name: logging-loki # <1>
  namespace: openshift-logging # <2>
spec:
  size: 1x.small # <3>
  storage:
    schemas:
    - version: v13
      effectiveDate: "<yyyy>-<mm>-<dd>" # <4>
    secret:
      name: logging-loki-s3 # <5>
      type: s3 # <6>
  storageClassName: <storage_class_name> # <7>
  tenants:
    mode: openshift-logging # <8>
----
<1> Use the name `logging-loki`.

> <2> You must specify `openshift-operators-redhat` as the namespace.

Reviewer 3: ```suggestion
<2> You must specify `openshift-logging` as the namespace.
```

Author: fixed

// Module is included in the following assemblies:
//
// * observability/logging/log_storage/installing-log-storage.adoc

:_mod-docs-content-type: PROCEDURE
[id="log6x-logging-loki-cli-install_{context}"]
= Installing the {loki-op} by using the CLI

> To install and configure logging on your {product-title} cluster, you must first install {loki-op}, which manages the log store `Loki`. `Loki` log store is deployed and configured by the {loki-op} by configuring the resource `LokiStack`. This can be done from the {product-title} command-line interface (CLI).

Reviewer: ```suggestion
To install and configure logging on your {product-title} cluster, you must first install {loki-op}, which manages the log store `Loki`. `Loki` log store is deployed and configured by the {loki-op} by reconciling the resource `LokiStack`. This can be done from the {product-title} command-line interface (CLI).
```

----

. Create a `Subscription` object for {loki-op}:
+
.Example `Subscription` object
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: loki-operator
  namespace: openshift-operators-redhat # <1>
spec:
  channel: stable-6.<y> # <2>
  installPlanApproval: Automatic # <3>
  name: loki-operator
  source: redhat-operators # <4>
  sourceNamespace: openshift-marketplace
----
<1> You must specify the `openshift-operators-redhat` namespace.
<2> Specify `stable-6.<y>` as the channel.
<3> If the approval strategy in the subscription is set to `Automatic`, the update process initiates as soon as a new Operator version is available in the selected channel. If the approval strategy is set to `Manual`, you must manually approve pending updates.
<4> Specify `redhat-operators` as the value. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the `CatalogSource` object that you created when you configured the Operator Lifecycle Manager (OLM).

. Apply the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

> . Create an `OperatorGroup` object

Reviewer 3: We should create `OperatorGroup` at first, then create the `Subscription`. Or the `Subscription` may raise error.

Author: done

+
.Example `OperatorGroup` object
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: loki-operator
  namespace: openshift-operators-redhat # <1>
  annotations:
    olm.providedAPIs: AlertingRule.v1.loki.grafana.com,LokiStack.v1.loki.grafana.com,RecordingRule.v1.loki.grafana.com,RulerConfig.v1.loki.grafana.com
spec:
  upgradeStrategy: Default
----
<1> You must specify the `openshift-operators-redhat` namespace.

. Apply the `OperatorGroup` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a `namespace` object to where we will deploy the `LokiStack`:
+
.Example `namespace` object
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: openshift-logging # <1>
  labels:
    openshift.io/cluster-monitoring: "true" # <2>
----
<1> `openshift-logging` is the namespace dedicated for all {logging} workloads.

> <2> A string value that specifies the label, as shown, to ensure that cluster monitoring scrapes the `openshift-operators-redhat` namespace.

Reviewer 3: ```suggestion
<2> A string value that specifies the label, as shown, to ensure that cluster monitoring scrapes the `openshift-logging` namespace.
```

Author: fixed

+
.Example `Secret` object
[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: logging-loki-s3 <1>
  namespace: openshift-logging
stringData: <2>
  access_key_id: AKIAIOSFODNN7EXAMPLE
  access_key_secret: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
  bucketnames: s3-bucket-name
  endpoint: https://s3.eu-central-1.amazonaws.com
  region: eu-central-1
----
<1> Use the name `logging-loki-s3` to match the name used in LokiStack.
<2> For the contents of the secret see the Loki object storage section.
+
--
include::snippets/logging-retention-period-snip.adoc[leveloffset=+1]
--

. Apply the `Secret` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a `LokiStack` CR:
+
.Example `LokiStack` CR
[source,yaml]
----
apiVersion: loki.grafana.com/v1
kind: LokiStack
metadata:
  name: logging-loki # <1>
  namespace: openshift-logging # <2>
spec:
  size: 1x.small # <3>
  storage:
    schemas:
    - version: v13
      effectiveDate: "<yyyy>-<mm>-<dd>" # <4>
    secret:
      name: logging-loki-s3 # <5>
      type: s3 # <6>

>       credentialMode:

Reviewer: No need to reference the `credentialMode` here. I missed this one
```suggestion
```

>       credentialMode:

Author: removed credentialMode

  name: cluster-logging
  source: redhat-operators # <3>
  sourceNamespace: openshift-marketplace
----
<1> You must specify the `openshift-logging` namespace.
<2> Specify `stable`, or `stable-6.<y>` as the channel.
<3> Specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the CatalogSource object you created when you configured the Operator Lifecycle Manager (OLM).

. Apply the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a `LokiStack` CR:
+
.Example `LokiStack` CR
[source,yaml]
----
apiVersion: loki.grafana.com/v1
kind: LokiStack
metadata:
  name: logging-loki # <1>
  namespace: openshift-logging # <2>
spec:
  size: 1x.small # <3>
  storage:
    schemas:
    - version: v13
      effectiveDate: "<yyyy>-<mm>-<dd>"
    secret:
      name: logging-loki-s3 # <4>
      type: s3 # <5>
      credentialMode: # <6>
  storageClassName: <storage_class_name> # <7>
  tenants:
    mode: openshift-logging # <8>
----
<1> Use the name `logging-loki`.
<2> You must specify the `openshift-logging` namespace.

> <3> Specify the deployment size. In the {logging} 5.8 and later versions, the supported size options for production instances of Loki are `1x.extra-small`, `1x.small`, or `1x.medium`.

Reviewer: Probably here we should also link the sizing page for further info. I'm talking about https://docs.openshift.com/container-platform/4.16/observability/logging/log_storage/installing-log-storage.html#loki-deployment-sizing_installing-log-storage (this linking part can be a follow up)
```suggestion
<3> Specify the deployment size. Supported size options for production instances of Loki are `1x.small`, `1x.extra-small`, `1x.small`, or `1x.medium`. In {logging} 6.1 we introduced `1x.pico`.
```
